### PR TITLE
bench: optionally record individual handshake latencies

### PR DESCRIPTION
**Pull request by ctz**

### CODE REVIEW

File: rustls-bench/src/main.rs [hunk 1]
```diff
@@ -268,13 +283,16 @@ fn bench_handshake(params: &Parameters) {
 }
 
 fn bench_handshake_buffered(
+    params: &Parameters,
```
Comment (djc) [inline rustls-bench/src/main.rs:292]: Argument order nit: take this last?

File: rustls-bench/src/main.rs [hunk 2]
```diff
@@ -1352,6 +1406,14 @@ impl TempBuffers {
     }
 }
 
+fn wall_time() -> f64 {
+    duration_nanos(
+        SystemTime::now()
```
Comment (djc) [inline rustls-bench/src/main.rs:1422]: Shouldn't this be using `Instant` instead?

File: rustls-bench/src/main.rs [hunk 3]
```diff
@@ -661,12 +667,17 @@ fn bench_memory(
     }
 }
 
-fn lookup_matching_benches(ciphersuite_name: &str) -> Vec<&BenchmarkParam> {
-    let r: Vec<&BenchmarkParam> = ALL_BENCHMARKS
+fn lookup_matching_benches(
+    ciphersuite_name: &str,
+    key_type: &Option<KeyType>,
```
Comment (djc) [inline rustls-bench/src/main.rs:672]: Since `KeyType` is `Copy` anyway, why take this by reference?

**Review by djc — APPROVED** [review]

File: rustls-bench/src/main.rs [hunk 4]
```diff
@@ -99,6 +100,13 @@ struct Args {
     )]
     multiplier: f64,
 
+    #[arg(
+        long,
+        env = "BENCH_LATENCY",
+        help = "Writes individual handshake latency into files starting with this string.  The files are named by appending a role (client/server), a thread id, and 'latency.tsv' to the given string."
+    )]
+    latency: Option<String>,
```
Comment (cpu) [inline rustls-bench/src/main.rs:108]: maybe this would be more descriptive as `latency-prefix`? :thinking:

**Review by cpu — APPROVED** [review]

File: rustls-bench/src/main.rs (hunk 2 above)
Comment (ctz) [reply rustls-bench/src/main.rs:1422]: I don't think you can go from an Instant to something that can be written a file? Aside from that, this is intended to produce similar output files to the C++ version which uses `gettimeofday` here.

Comment (djc) [reply rustls-bench/src/main.rs:1422]: Well, you can get a `Duration` by comparing two `Instant`s but I guess that doesn't help if you're trying to match C++ output.

**Pull request merged**